packages/document-service-api/src/controllers/documents.spec.js

```diff
@@ -504,4 +507,166 @@ describe('Documents controller', () => {
+      generateLocation: jest.fn(),
+      get: jest.fn(),
+      save: jest.fn().mockResolvedValue(),
+      validate: jest.fn().mockResolvedValue(),
+    };
+
+    doc.generateId.mockReturnValue(doc);
+
+    Documents.findOne.mockReturnValue(doc);
+    Documents.deleteOne.mockReturnValue({ deletedCount: 0 });
+
+    expect(await controller.deleteDocument(req, res)).toBe(undefined);
+
+    expect(res.status).toBeCalledWith(404);
+  });
+
+  it('Should return a 500 if delete from database fails', async () => {
+    const applicationId = 'some-app-id';
+    const documentId = 'some-doc-id';
+    const req = {
+      log: {
+        debug: jest.fn(),
+        error: jest.fn(),
+      },
+      params: {
+        applicationId,
+        documentId,
+      },
+    };
+
+    const doc = {
+      generateId: jest.fn(),
+      generateLocation: jest.fn(),
+      get: jest.fn(),
+      save: jest.fn().mockResolvedValue(),
+      validate: jest.fn().mockResolvedValue(),
+    };
+
+    doc.generateId.mockReturnValue(doc);
+
+    Documents.findOne.mockReturnValue(doc);
+    Documents.deleteOne.mockImplementation(() => {
+      throw new Error('Computer says no');
+    });
+
+    expect(await controller.deleteDocument(req, res)).toBe(undefined);
+
+    expect(res.status).toBeCalledWith(500);
+  });
+
+  it('Should return an error if delete from blob storage fails', async () => {
+    const applicationId = 'some-app-id';
+    const documentId = 'some-doc-id';
+    const req = {
+      log: {
+        error: jest.fn(),
+      },
+      params: {
+        applicationId,
+        documentId,
+      },
+    };
+
+    const doc = {
+      generateId: jest.fn(),
+      generateLocation: jest.fn(),
+      get: jest.fn(),
+      save: jest.fn().mockResolvedValue(),
+      validate: jest.fn().mockResolvedValue(),
+    };
+
+    doc.generateId.mockReturnValue(doc);
+
+    Documents.findOne.mockReturnValue(doc);
+    Documents.deleteOne.mockReturnValue({ deletedCount: 1 });
+    deleteFromBlobStorageByLocation.mockReturnValue(false);
+
+    expect(await controller.deleteDocument(req, res)).toBe(undefined);
+    expect(req.log.error).toHaveBeenCalled();
+    expect(res.status).toBeCalledWith(500);
+  });
+
+  it('Should return 204 when document deletion succeeds', async () => {
+    const applicationId = 'some-app-id';
+    const documentId = 'some-doc-id';
+    const req = {
+      log: {
+        debug: jest.fn(),
+        error: jest.fn(),
+      },
+      params: {
+        applicationId,
+        documentId,
+      },
+    };
+
+    const doc = {
+      generateId: jest.fn(),
+      generateLocation: jest.fn(),
+      get: jest.fn(),
+      save: jest.fn().mockResolvedValue(),
+      validate: jest.fn().mockResolvedValue(),
+    };
+
+    doc.generateId.mockReturnValue(doc);
+
+    Documents.findOne.mockReturnValue(doc);
+    Documents.deleteOne.mockReturnValue({ deletedCount: 1 });
+    deleteFromBlobStorageByLocation.mockReturnValue(true);
```

Comment:
think using `mockResolvedValue` might catch the `await` issue above

https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvaluevalue

same for reject